editoast/src/views/infra/auto_fixes/mod.rs

```diff
@@ -182,6 +206,116 @@ fn fix_infra(
     Ok(operations)
 }
 
+fn reduce_operation(op1: Option<Operation>, op2: Option<Operation>) -> Option<Operation> {
+    match (op1, op2) {
+        (Some(Operation::Create(_)), Some(Operation::Create(_)))
+        | (Some(Operation::Create(_)), Some(Operation::Delete(_)))
+        | (Some(Operation::Update(_)), Some(Operation::Create(_)))
+        | (Some(Operation::Delete(_)), Some(Operation::Create(_))) => {
+            error!("cannot reduce these 2 operations, it should never happen");
+            None
+        }
+        (Some(Operation::Create(railjson)), Some(Operation::Update(update))) => {
+            let UpdateOperation {
+                obj_id: id,
+                obj_type: r#type,
+                railjson_patch,
+                ..
+            } = update;
+            debug_assert_eq!(railjson.get_id(), &id);
+            debug_assert_eq!(railjson.get_type(), r#type);
+            serde_json::to_string(&*railjson)
+                .ok()
+                .and_then(|json| {
+                    let mut value = serde_json::from_str(&json).ok()?;
+                    json_patch::patch(&mut value, &railjson_patch).ok()?;
+                    Some(value)
+                })
+                .and_then(|value| serde_json::to_string(&value).ok())
+                .and_then(|json| serde_json::from_str(&json).ok())
```

Comment:
Feels like we could use [from_value](https://docs.rs/serde_json/latest/serde_json/fn.from_value.html)

Reply:
Didn't know about this API, thanks ! Done in https://github.com/osrd-project/osrd/pull/6012/commits/6cf4c0b79387a7c27bea4f426fc725d97b21dad3.